apis/v1alpha1/gateway_types.go

```diff
@@ -34,18 +34,25 @@ import (
 // This ensures that a GatewayClass associated with a Gateway(s) is not
 // deleted while in use.
 type Gateway struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	// Spec defines the desired state of Gateway.
+	//
+	// +optional
 	Spec GatewaySpec `json:"spec,omitempty"`
```

Comment:
I think it should be the other way around where we remove the `omitempty` tag. An object without a `spec` is going to confuse users and even maintainers.

Reply:
Are we OK diverging from how upstream defines spec/status/metadata optionality?

xref: https://github.com/kubernetes/api/blob/master/core/v1/types.go

Reply:
As strange as it is, I think it likely is appropriate to follow what upstream is doing here.